File: services/rust/poem/src/routes/middleware/log.rs
use poem::{Endpoint, IntoResponse, Request, Response, Result};
use tracing::{debug, instrument};

#[instrument(skip_all)]
pub async fn log_middleware<E: Endpoint>(next: E, req: Request) -> Result<Response> {
    debug!(method = %req.method(), path = %req.uri(), "Incoming request");

    next.call(req).await.map(IntoResponse::into_response)
}

File: services/typescript/hono/src/routes/middleware/log.ts
import type { Context, MiddlewareHandler, Next } from "hono";
import { getLogger } from "telemetry";

export const logMiddleware: MiddlewareHandler = (c: Context, next: Next) => {
  let method = c.req.method;
  let path = c.req.path;

  getLogger().debug({ method, path }, "Incoming request");

  return next();
};
